### DISCUSSION TITLE
Adafruit Bonnet has Defeated Me? (No Display.... At all...)

### DISCUSSION THREAD

**Discussion by taxicabyellow** [riffnshred/nhl-led-scoreboard#421] [Hardware]
using Raspi 4's, matrix(s) from Aliexpress (p5 2121-16s-6430 B0 / 2PS610c) and adafruit bonnets, i get zilch. i have 4 panels total, and 2 bonnets, have tried both bonnets on various boards, and i cannot get ANYTHING for output. I've compared pinout on my boards to the adafruit 64x32 p5 and they are identical. I've verified voltage out from bonnet is 4.8v. Have tried with pi powered seperatly as well. (does throw low power warning if not powered seperatly). have even tried numerous power and data cabels... Ideas where i can start here?

**Comment by riffnshred** [comment]
Are you using the image version or are you installing manually following the instructions in the wiki?

FYI, the pinouts on the panels can be the same, but the chips on them might be different, which might be the cause.

**Reply by taxicabyellow** [reply]
that makes sense... I used the image, which otherwise was flawless btw. Kudos. I did try the upgrade through ssh to .12 (i think). i'm at best getting blotches of color... Chips seem to say RUL6024

**Reply by riffnshred** [reply]
If you see boobs of colors or led flashing, make sure the Bonnets don't touch the raspberry pi underneath it. Also, make sure all cables are properly inserted in their socket.  I don't recognize the chip model but if you look at the [rpi-rgb-led-matrix](https://github.com/hzeller/rpi-rgb-led-matrix) repository, you might find some help. that's the driver we are using to run the display.

**Comment by taxicabyellow** [comment]
Anything to try with what i have on hand? or should i be buying the matrices direct from adafruit?

**Reply by riffnshred** [reply]
If what I posted dont work and you can't find answers on the display driver's repo, that would be the last resort yes. That's the thing with buying those displays on Aliexpress, it's basically a roll sometimes.

**Reply by yulefitzgivens** [reply]
I bought my panels from Wired Watts and they worked just fine and were cheaper than those at Adafruit. I am using 4 of the 64x32 P5 indoor panels. Hopes this help if you need new panels.

**Reply by taxicabyellow** [reply]
Did you use the indoor or outdoor matrices?

**Reply by yulefitzgivens** [reply]
I bought indoor panel since I was not going to put these outdoor and was getting them cheaper that outdoor panels. Here is a picture of mine, 4 panels.

![20230222_234914](https://github.com/riffnshred/nhl-led-scoreboard/assets/144061742/031bedbf-a6ba-45eb-a349-26f6acdd01ba)

**Reply by yulefitzgivens** [reply]
It does cut off some of the team logos as the code it not written for 4 panels.

**Reply by taxicabyellow** [reply]
This your code or utilizing the diy nhl scoreboard stuff for rpi? Looks great!

**Reply by rampar20** [reply]
how did you get it to use the four panels?

**Comment by yulefitzgivens** [comment]
I used the DIY NHL scoreboard stuff. I can post my command line parameters if you like.

**Reply by rampar20** [reply]
yes please

**Reply by yulefitzgivens** [reply]
Sorry for the lateness of my reply....

Here is my command line that I ended up with, I ran mine using Supervisor. 

command=sudo python3 src/main.py --led-gpio-mapping=adafruit-hat-pwm --led-brightness=60 --led-slowdown-gpio=2  --led-chain=4  --led-parallel=1 --led-pixel-mapper=U-mapper  --led-cols=64 --led-rows=32

**Comment by rampar20** [comment]
Thank you. Will give it a try when I get back from my son's hockey
tournament. Seems like I did add this to mine to make it work,
(--led-pixel-mapper=U-mapper), well at least I am hopeful.
